src/algebra/category/FinVect.lean

```diff
@@ -0,0 +1,194 @@
+/-
+Copyright (c) 2021 Jakob von Raumer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob von Raumer
+-/
+import linear_algebra.finite_dimensional
+import category_theory.monoidal.rigid
+import linear_algebra.finsupp_vector_space
+import linear_algebra.dual
+import linear_algebra.coevaluation
+import .Module.monoidal
+
+/-!
+# The category of finite dimensional vector spaces
+
+This introduces `FinVect K`, the category of finite dimensional vector spaces on a field `K`.
+It is implemented as a full subcategory on a subtype of  `Module K`.
+We first create the instance as a category, then as a monoidal category and then as a rigid monoidal
+category.
+
+## Future work
+
+* Show that `FinVect K` is a symmetric monoidal category.
+
+-/
+noncomputable theory
+
+open category_theory Module.monoidal_category
+open_locale classical big_operators
+
+universes u
+
+variables (K : Type u) [field K]
+
+/-- Define `FinVect` as the subtype of `Module.{u} K` of finite dimensional vector spaces. -/
+def FinVect := { V : Module.{u} K // finite_dimensional K V.carrier }
+
+namespace FinVect
+
+instance category : category (FinVect K) := by unfold FinVect; apply_instance
+
+instance finite_dimensional (V : FinVect K): finite_dimensional K V.val := V.prop
+
+instance has_sort_coe : has_coe_to_sort (FinVect K) := ⟨_, λ V, V.val⟩
+
+instance has_coe_to_fn (V W : FinVect K) : has_coe_to_fun (V ⟶ W) :=
+  ⟨λ _, V.val → W.val, λ f, f.to_fun⟩
+
+instance : inhabited (FinVect K) := ⟨⟨Module.of K K, finite_dimensional.finite_dimensional_self K⟩⟩
+
+/- This should go to `linear_algebra` after `linear_algebra` is cleaned up.
+  Right now it cannot go to either `linear_algebra.tensor_product` or
+  `linear_algebra.finite_dimension` because of dependencies. -/
+instance finite_dimensional_tensor_product (V V₂ : Type*) [add_comm_group V]
+  [module K V] [add_comm_group V₂] [module K V₂]
+  [finite_dimensional K V] [finite_dimensional K V₂] :
+  finite_dimensional K (tensor_product K V V₂) :=
+finite_dimensional.of_fintype_basis
+  (finsupp.basis.tensor_product (basis.of_vector_space K V) (basis.of_vector_space K V₂))
+
+instance monoidal_category : monoidal_category (FinVect K) :=
+@monoidal_category.full_monoidal_subcategory _ _ Module.monoidal_category
+ (λ (V : Module K), finite_dimensional K V)
+ (by { exact finite_dimensional.finite_dimensional_self K})
+ (λ X Y hX hY, @FinVect.finite_dimensional_tensor_product K _ X Y _ _ _ _ hX hY)
+
+-- This should go to `linear_algebra`.
```

Comment:
There's no obstacle to doing so now?

Reply:
Even better, found the instance :facepalm: